exec/csnex/Global.cs

```diff
@@ -68,13 +68,195 @@ public void array__extend()
             }
         }
 
+        public void array__find()
+        {
+            Cell e = Exec.stack.Pop();
+            List<Cell> a = Exec.stack.Pop().Array;
+
+            Number r = new Number(-1);
+
+            for (int i = 0; i < a.Count; i++) {
+                if (a[i].Equals(e)) {
+                    r = new Number(i);
+                    break;
+                }
+            }
+
+            if (r.IsNegative()) {
+                Exec.Raise("ArrayIndexException", "value not found in array");
+                return;
+            }
+
+            Exec.stack.Push(new Cell(r));
+        }
+
+        public void array__range()
+        {
+            Number step = Exec.stack.Pop().Number;
+            Number last = Exec.stack.Pop().Number;
+            Number first = Exec.stack.Pop().Number;
+            List<Cell> r = new List<Cell>();
+
+            if (step.IsZero()) {
+                Exec.Raise("ValueRangeException", step.ToString());
+                return;
+            }
+
+            if (step.IsNegative()) {
+                for (Number i = first; Number.IsGreaterOrEqual(i, last); i = Number.Add(i, step)) {
+                    r.Add(new Cell(i));
+                }
+            } else {
+                for (Number i = first; Number.IsLessOrEqual(i, last); i = Number.Add(i, step)) {
+                    r.Add(new Cell(i));
+                }
+            }
+
+            Exec.stack.Push(new Cell(r));
+        }
+
+        public void array__remove()
+        {
+            Number index = Exec.stack.Pop().Number;
+            List<Cell> array = Exec.stack.Pop().Address.Array;
+
+            if (!index.IsInteger()) {
+                Exec.Raise("ArrayIndexException", index.ToString());
+                return;
+            }
+
+            int i = Number.number_to_int32(index);
+            array[i].ResetCell();
```

Comment:
You probably need some range checking here.

Reply:
Checking against what?  The compiler would never generate code that would violate this, would it?

Reply:
It's not about code generation. What if `i` is out of range of `array`?

Reply:
Made range checking match global.cpp.

Reply:
> It's not about code generation. What if `i` is out of range of `array`?

Yea.  I considered that, and so I copied the range checking out of global.cpp.